### DISCUSSION TITLE
MembershipListener not triggering after a while (jGroups 4.0)

### DISCUSSION THREAD

**Discussion by SilvanaP** [belaban/JGroups#607] [Q&A]
Hello, I am using jGroups in a setup where I have one "master" like member which always stays in the group and other members can join or leave this group arbitrarily. A custom membership listener (which extends MembershipListener) in the "master" memebr keeps track of the changes in the group and logs to the console when a member joins or leaves the group.

This works fine for a while (20-30 mins on average), but then suddenly, the membership listener is not triggered anymore when a memeber joins or leaves. The members themselves show in their log output that they successfully connected to a group, though. 
The "master" memeber does not show an error or whatsoever and is running without a problem, so I am not sure if either I am using jGroups in a wrong way or if the master is kicked out of the group silently and never reconnects?

Did anyone here have similar problems with jGroups? Any hints or even suggestions for a fix would be highly appreciated.

**Comment by belaban** [comment]
Do you have a reproducer?

**Reply by SilvanaP** [reply]
@belaban What do you mean by reproducer? Is this a Class in jGroups? I never came across the word in the docs... 🫣

**Comment by belaban** [comment]
I meant a sample program which reproduces the problemBela Ban-------- Original message --------From: SilvanaP ***@***.***>Date: Mon, 28 Feb 2022, 19:33To: belaban/JGroups ***@***.***>Cc: Bela Ban ***@***.***>, Mention ***@***.***>Subject: Re: [belaban/JGroups] MembershipListener not triggering after a while (jGroups 4.0) (Discussion #607)

@belaban What do you mean by reproducer? Is this a Class in jGroups? I never came across the word in the docs... 🫣

**Reply by SilvanaP** [reply]
🤦 Sorry, how could I not get this ...
Unfortunately no, the project I have is quite complex but I could try to prepare something and post the code tomorrow.

Do you have any ideas what the problem could be, or did you never hear of something similar? I suspect that my "master" member disconnects silently from the group and gets no more updates? Is there any way to get a my status in the group so I can re-connect eventually? As far as I understood, this should happen automatically via a "merge", but apparently it doesn't.

**Comment by belaban** [comment]
No idea. If the master disconnects, then it won't get any more updates, but you should see a view change in that case... You can use probe.sh to look at all members of the cluster, including their views.
If you do get view changes, but they stop after some time, then that points to a disconnect. As I said, only a stripped-down version of your code, which shows the issue (including configuration) will let me take a stab at solving it...

**Reply by SilvanaP** [reply]
Ok, I will inspect further tomorrow, and if possible, I will provide a min. sample. In case I manage to fix the problem, I will keep you posted, too. Thanks for the super-quick reply!
